### PR TITLE
Consolidate recommender / strat user guides and examples

### PR DESCRIPTION
**Pull request by Scienfitz**
This is based on #147 

This rewrites and moves parts of the user guides and refactored further some examples. This is NOT intended as a complete rewrite or additional of much content, but just a consolidation due to the changes that the recommender/strategy refactoring brought with them

### CODE REVIEW

File: examples/Basics/Basics_Header.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 These examples demonstrate the most basic aspects of BayBE: How to set up a
 {doc}`Campaign </userguide/campaigns>` and how to configure an optimization
-{doc}`Strategy </userguide/strategies>`.
\ No newline at end of file
+{doc}`Recommenders </userguide/recommenders>`.
```
Comment (AVHopp) [inline examples/Basics/Basics_Header.md:5]: Would use singular here as the sentence is "How to configure an optimization recommender".

File: streamlit/initial_strategy.py [hunk 2]
```diff
@@ -73,7 +73,9 @@ def main():
 
     # simulation parameters
     random_seed = int(st.sidebar.number_input("Random seed", value=42))
-    strategy_name = st.sidebar.selectbox("Strategy", list(selection_strategies.keys()))
+    strategy_name = st.sidebar.selectbox(
```
Comment (AVHopp) [inline streamlit/initial_strategy.py:76]: Should this still say "Strategy"? Ad should maybe the whole file also be renamed to `initial_recommender`?

File: docs/userguide/recommenders.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+recommender = TwoPhaseMetaRecommender(
+    initial_recommender=RandomRecommender(), recommender=SequentialGreedyRecommender()
+)
+```
+
+* The **[`SequentialMetaRecommender`](baybe.recommenders.meta.sequential.SequentialMetaRecommender)**
+  introduces a simple yet versatile approach by utilizing a predefined list of
+  recommenders. By specifying the desired behavior using the `mode` attribute, it is
+  possible to flexibly determine the strategy's response when it exhausts the available
```
Comment (AVHopp) [inline docs/userguide/recommenders.md:105]: Double-check if this "strategy" should be replaced by "meta recommender"

**Review by AVHopp — APPROVED** [review]
Looks good, there are just some parts where you should double check again.

File: streamlit/initial_strategy.py (hunk 2 above)
Comment (Scienfitz) [reply streamlit/initial_strategy.py]: changed

File: examples/Basics/Basics_Header.md (hunk 1 above)
Comment (Scienfitz) [reply examples/Basics/Basics_Header.md:5]: fixed

File: docs/userguide/recommenders.md (hunk 3 above)
Comment (Scienfitz) [reply docs/userguide/recommenders.md]: fixed

**Pull request merged**